moshi/src/main/java/com/squareup/moshi/JsonWriter.java

```diff
@@ -360,12 +360,32 @@ public final void setSerializeNulls(boolean serializeNulls) {
 
   /**
    * Writes {@code source} directly without encoding its contents.
-   * Since no validation is performed, {@link #setSerializeNulls} and other writer configurations
-   * are not respected.
    *
-   * @return this writer.
+   * @see #valueSink()
+   * @deprecated Use: {@code try (BufferedSink sink = writer.valueSink()) { source.readAll(sink): }}
```

Comment:
I wouldn’t bother to deprecate; we’ll never remove it and it just punishes users for no reason

Reply:
(We made a mistake by offering this bad API, but our users don’t need to adjust anything because there’s no reason we’d ever want to remove this.)